files/default/slurm/generate_slurm_config.py

```diff
@@ -0,0 +1,104 @@
+import json
+from os import makedirs, path
+
+import argparse
+
+from jinja2 import Environment, FileSystemLoader
+from jsonschema import validate
```

Comment:
although jsonschema is useful here I would drop all non-necessary dependencies on third party libraries.